src/logging.c

```diff
@@ -121,9 +137,14 @@ rcutils_ret_t rcutils_logging_shutdown()
   }
   rcutils_ret_t ret = RCUTILS_RET_OK;
   if (g_rcutils_logging_severities_map_valid) {
-    rcutils_ret_t ret = rcutils_string_map_fini(&g_rcutils_logging_severities_map);
-    if (ret != RCUTILS_RET_OK) {
-      fprintf(stderr, "Failed to finalize logging severities map: return code %d", ret);
+    rcutils_ret_t string_map_ret = rcutils_string_map_fini(&g_rcutils_logging_severities_map);
+    if (string_map_ret != RCUTILS_RET_OK) {
+      fprintf(
+        stderr,
+        "Failed to finalize logging severities map: %s\n", rcutils_get_error_string_safe());
+      rcutils_reset_error();
```

Comment:
Same thing here, I'd prefer to integrate this error into the new error message and not print anything to stderr.